### PR TITLE
1)InvokeCall added, instead of all different …

### PR DESCRIPTION
**Pull request by Vizit0r**
...callers (x86, x64, powerpc etc) for Delphi 2010+.

Tested on Win x86&x64, Android, MacOS32 - no problems observed.

2) Changes for correct MACOS compilation in Delphi

3) few changes and fixes for correct PS work on D7.

### CODE REVIEW

File: Source/uPSComponent.pas [hunk 1]
```diff
@@ -4,7 +4,9 @@ interface
 
 uses
   SysUtils, Classes, uPSRuntime, uPSDebugger, uPSUtils,
-  uPSCompiler, uPSC_dll, uPSR_dll, uPSPreProcessor;
+  uPSCompiler,
+  {$IF DEFINED (MSWINDOWS) OR Defined (UNIX) OR Defined (fpc)} uPSC_dll, uPSR_dll,{$IFEND}
```
Comment (carlokok) [inline Source/uPSComponent.pas]: This is still here? AFAIK this doesn't compile in d2..d7.

Comment (Vizit0r) [reply Source/uPSComponent.pas]: compiled in D7 normally.
About previous version have no idea - they cant be installed on Win7 x64.

Comment (Vizit0r) [reply Source/uPSComponent.pas]: Screenshot_1
Screen for you from D7 :)

**Comment by Vizit0r** [comment]
so, Carlo?

next pack awaiting :)

File: Source/uPSDebugger.pas [hunk 2]
```diff
@@ -531,7 +531,7 @@ function TPSCustomDebugExec.GetCallStack(var Count: Cardinal): tbtString;
       else
         Result:= Result + ParamList.Items[I] + ': ' +
                  PSVariantToString(NewTPSVariantIFC(FStack[Cardinal(Longint(StackBase) - Longint(I) - 1)], False), '') + '; ';
-    Result := tbtString(String(Result).Remove(Length(Result)-2));
+ //   Result := tbtString(String(Result).Remove(Length(Result)-2));
```
Comment (martijnlaan) [inline Source/uPSDebugger.pas]: @Vizit0r Why is this line commented now?

**Comment by martijnlaan** [comment]
@carlokok @Vizit0r and others:

Ever since this pull request got merged it seems that ROPS has gotten unstable. There have been quite some issues reported with it, some with merged fixes (#211, #213) but also some without merged fixes (#217, #218, #219). Now I don't really know if it's safe to update anymore and am considering to create a personal fork at 1f846a56c81997b5cafe20d8dbd2234a88314a63, which is the commit before @Vizit0r's two big merged pull requests (this one and #205).

In hindsight, were these pull requests worth it? If not, would it be better to revert them?

It doesn't help that both pull requests contain various different and unrelared changes, making it much harder to check things.

**Comment by martijnlaan** [comment]
@Vizit0r @carlokok 
I talked to Carlo and as he suggested added a PS_USECLASSICINVOKE which makes it possible to avoid InvokeCall.inc but this still leaves a bunch of unexplained changes. For example in #205 I count 4 unexplained changes.

**Comment by Vizit0r** [comment]
sorry gentlemens, i've been on work last 3,5 month with internet speed near zero.
Will read, analyze and answer soon.

**Pull request merged**